include/RAJA/pattern/WorkGroup/WorkStruct.hpp

```diff
@@ -71,7 +71,8 @@ struct WorkStruct<size, Dispatcher<platform, dispatch_policy, DispatcherID, Call
         "WorkStruct and GenericWorkStruct must have obj at the same offset");
     static_assert(sizeof(value_type) <= sizeof(true_value_type),
         "WorkStruct must not be smaller than GenericWorkStruct");
-
+    static_assert(16 >= alignof(std::max_align_t) && (16/alignof(std::max_align_t))*alignof(std::max_align_t) == 16,
+			  "WorkStruct max align value must be 16 bytes");
```

Comment:
I think we should put this in the config so we ensure RAJA_MAX_ALIGN is valid everywhere. This should use RAJA_MAX_ALIGN instead of 16.